src/treecodes.py

```diff
@@ -1,238 +1,58 @@
-             'web-platform-tests-e10s-6': {'group': 'W-e10s', 'code': '6'},
-             'web-platform-tests-e10s-7': {'group': 'W-e10s', 'code': '7'},
-             'web-platform-tests-e10s-8': {'group': 'W-e10s', 'code': '8'},
-             'web-platform-tests-reftests-e10s': {'group': 'W-e10s', 'code': 'R'},
-             'mochitest-jetpack': {'group': '', 'code': 'JP'},
-             'luciddream': {'group': '', 'code': 'Ld'},
-             'robocop-1': {'group': 'RC', 'code': '1'},
-             'robocop-2': {'group': 'RC', 'code': '2'},
-             'robocop-3': {'group': 'RC', 'code': '3'},
-             'robocop-4': {'group': 'RC', 'code': '4'},
-             'robocop-5': {'group': 'RC', 'code': '5'},
-             'robocop-6': {'group': 'RC', 'code': '6'},
-             'robocop-7': {'group': 'RC', 'code': '7'},
-             'robocop-8': {'group': 'RC', 'code': '8'},
-             'reftest': {'group': 'R', 'code': 'R'},
-             'tp5o': {'group': 'T', 'code': 'tp'},
-             'tp5o-e10s': {'group': 'T-e10s', 'code': 'tp'},
-             'tp5o-osx-e10s': {'group': 'T-e10s', 'code': 'tp'},
-             'other': {'group': 'T', 'code': 'o'},
-             'other-e10s': {'group': 'T-e10s', 'code': 'o'},
-             'other_nol64': {'group': 'T', 'code': 'o'},
-             'other-e10s_nol64': {'group': 'T-e10s', 'code': 'o'},
-             'other_l64': {'group': 'T', 'code': 'o'},
-             'other-e10s_l64': {'group': 'T-e10s', 'code': 'o'},
-             'svgr': {'group': 'T', 'code': 's'},
-             'svgr-e10s': {'group': 'T-e10s', 'code': 's'},
-             'svgr-osx-e10s': {'group': 'T-e10s', 'code': 's'},
-             'g1': {'group': 'T', 'code': 'g1'},
-             'g1-e10s': {'group': 'T-e10s', 'code': 'g1'},
-             'g1-osx-e10s': {'group': 'T-e10s', 'code': 'g1'},
-             'g2': {'group': 'T', 'code': 'g2'},
-             'g2-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'g2-osx-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'dromaeojs': {'group': 'T', 'code': 'd'},
-             'dromaeojs-e10s': {'group': 'T-e10s', 'code': 'd'},
-             'chromez': {'group': 'T', 'code': 'c'},
-             'chromez-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'chromez-osx-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'xperf': {'group': 'T', 'code': 'x'},
-             'xperf-e10s': {'group': 'T-e10s', 'code': 'x'},
-             'androidx86-set-4': {'group': 'S', 'code': '4'}}
+import requests
 
 
-def getGroup(name):
-    try:
-        group = name.split('-')[0]
-    except:
-        group = ''
-    return group
+class Treecodes:
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        url = 'https://treeherder.mozilla.org/api/project/%s/runnable_jobs/?format=json'
+        data = requests.get(url % repo_name).json()
+        joblist = data['results']
```

Comment:
Remember you could add redo support to help for flaky connections:
https://hg.mozilla.org/hgcustom/version-control-tools/rev/fb8e1405d593

I'm not requirying it for this change but if you see issues in the alerting system you could lean on this.